SPEC.rdoc

```diff
@@ -260,31 +279,63 @@ The values of the header must be Strings,
 consisting of lines (for multiple header values, e.g. multiple
 <tt>Set-Cookie</tt> values) separated by "\\n".
 The lines must not contain characters below 037.
+
 === The Content-Type
+
 There must not be a <tt>Content-Type</tt>, when the +Status+ is 1xx,
 204 or 304.
+
 === The Content-Length
+
 There must not be a <tt>Content-Length</tt> header when the
 +Status+ is 1xx, 204 or 304.
+
 === The Body
-The Body must respond to +each+
-and must only yield String values.
 
-The Body itself should not be an instance of String, as this will
-break in Ruby 1.9.
+The Body is typically an Array of Strings, the application instance
+itself, or a File-like object.
+
+The Body must respond to +each+ or +call+. It may optionally respond to
++to_path+.
+
+A Body that responds to +each+ is considered to be a chunked body. If
+it also responds to +to_path+ is considered to be the same as that file
+the local storage.
+
+A Body that responds to +call+ is considered to be a streaming body.
 
-If the Body responds to +close+, it will be called after iteration. If
-the body is replaced by a middleware after action, the original body
-must be closed first, if it responds to close.
+If the Body responds to +close+, it will be called after iteration.
+If the body is replaced by a middleware after action, the original
+body must be closed first, if it responds to close.
+
+==== Chunked Body
```

Comment:
Maybe we should call this "Enumerable Body" rather than "Chunked Body"?  "Chunk" seems like a heavily used term with various meanings in HTTP, and really we want to communicate to users that the body needs to respond to `each`

Reply:
Completely agree, will update the PR.

Reply:
Sorry @SamSaffron I did not realise you were referring to this, it only just now clicked.

Reply:
Updated.